File: CS/PersistentObjects.cs
using System;
using DevExpress.Xpo;

namespace XpoEditForm {
    public class Order : XPObject {
        public Order(Session session) : base(session) { }

        string fCustomer;
        public string Customer {
            get { return fCustomer; }
            set { SetPropertyValue("Customer", ref fCustomer, value); }
        }

        DateTime fOrderDate;
        public DateTime OrderDate {
            get { return fOrderDate; }
            set { SetPropertyValue("OrderDate", ref fOrderDate, value); }
        }

        [Association("Order-Details")]
        public XPCollection<OrderDetail> OrderDetails { get { return GetCollection<OrderDetail>("OrderDetails"); } }
    }

    public class OrderDetail : XPObject {
        public OrderDetail(Session session) : base(session) { }

        string fProductName;
        public string ProductName {
            get { return fProductName; }
            set { SetPropertyValue("ProductName", ref fProductName, value); }
        }

        DateTime fOrderDate;
        public DateTime OrderDate {
            get { return fOrderDate; }
            set { SetPropertyValue("OrderDate", ref fOrderDate, value); }
        }

        int fQuantity;
        public int Quantity {
            get { return fQuantity; }
            set { SetPropertyValue("Quantity", ref fQuantity, value); }
        }

        decimal fPrice;
        public decimal Price {
            get { return fPrice; }
            set { SetPropertyValue("Price", ref fPrice, value); }
        }

        Order fOrder;
        [Association("Order-Details")]
        public Order Order {
            get { return fOrder; }
            set { SetPropertyValue("Order", ref fOrder, value); }
        }
    }
}

File: VB/PersistentObjects.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Namespace XpoEditForm
	Public Class Order
		Inherits XPObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		Private fCustomer As String
		Public Property Customer() As String
			Get
				Return fCustomer
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Customer", fCustomer, value)
			End Set
		End Property

		Private fOrderDate As DateTime
		Public Property OrderDate() As DateTime
			Get
				Return fOrderDate
			End Get
			Set(ByVal value As DateTime)
				SetPropertyValue("OrderDate", fOrderDate, value)
			End Set
		End Property

		<Association("Order-Details")> _
		Public ReadOnly Property OrderDetails() As XPCollection(Of OrderDetail)
			Get
				Return GetCollection(Of OrderDetail)("OrderDetails")
			End Get
		End Property
	End Class

	Public Class OrderDetail
		Inherits XPObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		Private fProductName As String
		Public Property ProductName() As String
			Get
				Return fProductName
			End Get
			Set(ByVal value As String)
				SetPropertyValue("ProductName", fProductName, value)
			End Set
		End Property

		Private fOrderDate As DateTime
		Public Property OrderDate() As DateTime
			Get
				Return fOrderDate
			End Get
			Set(ByVal value As DateTime)
				SetPropertyValue("OrderDate", fOrderDate, value)
			End Set
		End Property

		Private fQuantity As Integer
		Public Property Quantity() As Integer
			Get
				Return fQuantity
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue("Quantity", fQuantity, value)
			End Set
		End Property

		Private fPrice As Decimal
		Public Property Price() As Decimal
			Get
				Return fPrice
			End Get
			Set(ByVal value As Decimal)
				SetPropertyValue("Price", fPrice, value)
			End Set
		End Property

		Private fOrder As Order
		<Association("Order-Details")> _
		Public Property Order() As Order
			Get
				Return fOrder
			End Get
			Set(ByVal value As Order)
				SetPropertyValue("Order", fOrder, value)
			End Set
		End Property
	End Class
End Namespace
